bigtable/spark/src/main/scala/example/DataFrameDemo.scala

```diff
@@ -0,0 +1,92 @@
+package example
+
+import org.apache.hadoop.hbase.spark.datasources.{HBaseSparkConf, HBaseTableCatalog}
+
+import scala.util.Try
+
+object DataFrameDemo extends App {
+
+  println("Starting up...")
+
+  val projectId = args(0)
+  val instanceId = args(1)
+  val table = Try(args(2)).getOrElse("dataframe-demo")
+  val numRecords = Try(args(3).toInt).getOrElse(10)
+
+  import org.apache.spark.sql.SparkSession
+  val spark = SparkSession.builder().master("local[*]").getOrCreate()
+  println("Spark version: " + spark.version)
+
+  val catalog =
+    s"""{
+       |"table":{"namespace":"default", "name":"$table", "tableCoder":"PrimitiveType"},
+       |"rowkey":"key",
+       |"columns":{
+       |"col0":{"cf":"rowkey", "col":"key", "type":"string"},
+       |"col1":{"cf":"cf1", "col":"col1", "type":"boolean"},
+       |"col2":{"cf":"cf2", "col":"col2", "type":"double"},
+       |"col3":{"cf":"cf3", "col":"col3", "type":"int"}
+       |}
+       |}""".stripMargin
+  println(s"Writing $numRecords records to $table")
+
+  // FIXME Explain the options
+  // FIXME Where to find more options supported? Any docs?
+  val opts = Map(
+    HBaseTableCatalog.tableCatalog -> catalog,
+    HBaseTableCatalog.newTable -> "5",
+    HBaseSparkConf.USE_HBASECONTEXT -> "false") // accepts xml configs only
+
+  import spark.implicits._
+  val records = (0 until numRecords).map(BigtableRecord.apply).toDF
+
+  // TODO Use a command-line option to switch between command line params and xml
+
+  // Hack to specify HBase properties on command line
+  // BEGIN
+  // import org.apache.hadoop.hbase.HBaseConfiguration
+  // val conf = HBaseConfiguration.create()
+  // conf.set("google.bigtable.project.id", projectId)
+  // conf.set("google.bigtable.instance.id", instanceId)
+  import com.google.cloud.bigtable.hbase.BigtableConfiguration
+  val conf = BigtableConfiguration.configure(projectId, instanceId)
+
+  import org.apache.hadoop.hbase.spark.HBaseContext
+  val hbaseContext = new HBaseContext(spark.sparkContext, conf)
+  val opts_nouse = opts.filterNot { case (k, _) => k == HBaseSparkConf.USE_HBASECONTEXT }
```

Comment:
I'm confused by this line:
- why is it separate from the opts definition above?
- why is the option being removed rather then reset to true?

Reply:
This is a mixture of two "conflicting" approaches to hbase/bigtable configuration - command line vs xml file. The whole block is to support configuration on command line while the comment a few lines above (that I quoted below) shows how to do it with hbase-site.xml config.

```
HBaseSparkConf.USE_HBASECONTEXT -> "false") // accepts xml configs only
```